src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -191,16 +193,21 @@ export default class DefaultMessagingSession implements MessagingSession {
     }
   }
 
+  private retryConnection(): boolean {
+    return this.reconnectController.retryWithBackoff(async () => {
+      try {
+        await this.startConnecting(true);
+      } catch (e) {
+        // getMessagingSessionEndpoint can fail, so need to retry here
+        this.retryConnection();
```

Comment:
I think `this.retryConnection()` needs to return a boolean value to check whether to trigger a reconnection or not right?

Something like:
```
private retryConnection(): boolean {
  try {
   if (this.reconnectController.retryWithBackoff(async () => await this.startConnecting(true))) {
     return true;
   } else {
     return false;
   }
  } catch (error) {
    this.logger.error(`${error}`);
    return retryConnection();
  }
}
```